app/models/audit/event/authn.rb

```diff
@@ -81,8 +84,14 @@ def attempted_action
       def auth_stuctured_data
         { authenticator: @authenticator_name }.tap do |sd|
           sd[:service] = service_id if @service
+          sd[:user] = username
         end
       end
+
+      # In case of success returns the role performing the action, otherwise conceals it for security concerns
+      def username
```

Comment:
this is actually the role id and not the username. for example:
role_id: `cucumber:user:alice`
username: `alice`.

I would change this method name to something like `validated_role_id` or remove this method and move the logic to line 87. whatever you think is best.